tools/clang/test/HLSLFileCheck/hlsl/control_flow/loops/struct_exit_inner_latch_regression.ll

```diff
@@ -0,0 +1,75 @@
+; RUN: %opt %s -S -loop-unroll,StructurizeLoopExits=1 | FileCheck %s
+; RUN: %opt %s -S -dxil-loop-unroll,StructurizeLoopExits=1 | FileCheck %s
```

Comment:
These are `dxopt` command lines not `opt`. Can we try putting this test somewhere that it runs with LIT instead of the legacy infrastructure?

Reply:
I don't think opt can run passes with arguments like `StructurizeLoopExits`. These options are off by default.

Reply:
This isn’t using a `cl::opt` option, it is running with the `dxopt` option handling. I’m pretty sure that’s what the python script generates.

Reply:
Yeah it's using the `applyOptions` function added to the pass. the hctdb.py must be generating the `build\DXC\lib\HLSL\DxcOptimizer.inc` file that gets included in `DxcOptimizer.cpp`.

Reply:
I don't want to hold this change up, but I'm unsure why you marked this as resolved.

When run in the `HLSLFileCheck` folder `%opt` is actually `dxopt` not LLVM's `opt`. I'm pretty sure we can move this test into the `tools/clang/test/DXC/*` folder and change the `RUN` lines to `%dxopt`, and they should work just fine.

That makes this test run through LIT, using LIT's test scheduler which better distributes test execution.

Reply:
Oh I see what you mean now. Change incoming.

Reply:
Done.